app/ptokens/models.py

```diff
@@ -120,4 +129,47 @@ def save(self, *args, **kwargs):
 
     @property
     def title(self):
-        return self.title
\ No newline at end of file
+        return self.title
+
+
+class RedemptionToken(SuperModel):
+    """Define the structure of a Redemption PToken"""
+
+    REDEMPTION_STATUS_CHOICES = [
+        ('request', 'requested'),
+        ('accepted', 'accepted'),
+        ('denied', 'denied'),
+        ('completed', 'completed')
+    ]
+    ptoken = models.ForeignKey(PersonalToken, null=True, on_delete=models.SET_NULL)
+    redemption_state = models.CharField(max_length=50, choices=REDEMPTION_STATUS_CHOICES, default='request', db_index=True)
+    network = models.CharField(max_length=255, default='')
+    reason = models.CharField(max_length=1000, default='')
+    total = models.DecimalField(default=0, decimal_places=2, max_digits=50, blank=True, null=True, help_text='Total ptokens to redeem')
+    txid = models.CharField(max_length=255, default='')
```

Comment:
txid are not mandatory to create a Redemption?

Reply:
No, it is required when the holder accept the redemption, in such moment it transfer the tokens to the token owner.

Reply:
@zoek1 the transfer occurs when the Buyer (sender) marks the redemption as Complete. There is no web 3 transfer of tokens at the Accept/Deny status of the redemption, just database update.